### PR TITLE
T14: Build the reset password form & T15: Build the change username form

### PR DESCRIPTION
**Pull request by jonathan-952**
I added the reset password form and the reset username form. I started creating a sidebar that isn't complete but has the functionality, with buttons routing to the forms attached. 
image
image
image

### CODE REVIEW

**Comment by kdata12** [comment]
Hi! I’m glad to see you’ve successfully created a PR by yourself! To make this PR even better, there are 3 key items that we need to include:

- A PR title: in your case, it should "T14: Build the reset password form & T15: Build the change username form"
- A brief description of your work, what changes have been made, and any important details that would help the reviewer understand your work
- A screenshot of your work (only for frontend) so we can see what you did visually

**Review by kdata12 — COMMENTED** [review]
This is looking good

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,24 +1,17 @@
-import logo from './logo.svg';
 import './App.css';
+// import ResetPasswordForm from './resetPswd.js'
+// import ResetUsernameForm from './resetUsername.js';
```
Comment (kdata12) [inline frontend/src/App.js]: remove unnecessary comments here

File: frontend/src/formPage.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import './form.css'
+// import ResetPasswordForm from './resetPswd.js'
+// import ResetUsernameForm from './resetUsername.js'
```
Comment (kdata12) [inline frontend/src/formPage.js]: nit: remove comments

File: frontend/src/form.css [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    border: none;
+    font-size: 14px;
+    padding-left: 10px;
+}
+
+h1 {
+    font-size: 30px;
+}
+
+.nav_bar {
```
Comment (kdata12) [inline frontend/src/form.css]: not sure if this `nav_bar` classname refers to the actual nav_bar on the webpage, this name is kinda confusing to the reader.

File: frontend/src/resetPswd.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            height = {size}
+        />
+        </div>
+    );
+}
+
+export function Textbox({name, type, display_name}) {
+    return (
+        <div className = "textbox">
+            <input type = {type} name = {name} id = {name} placeholder = {display_name}/>
```
Comment (kdata12) [inline frontend/src/resetPswd.js:29]: consider adding a label element here like this

```css
<label htmlFor={name}>{display_name}</label>
<input type = {type} name = {name} id = {name} placeholder = {display_name}/>  
```

**Review by kdata12 — COMMENTED** [review]
I like the way you use different components like the TextBox and FormButton to build out your `ResetUsernameForm` and `ResetPasswordForm`! But do understand that it will introduce some coupling in your code where if you were to modify the `TextBox` component, it will cascade the changes down to the `ResetUsernameForm` and `ResetPasswordForm` components.  Just something to keep in mind.

File: frontend/src/resetPswd.js (hunk 4 above)
Comment (jonathan-952) [reply frontend/src/resetPswd.js:29]: i did this inteionally because i think it looks better

Comment (kdata12) [reply frontend/src/resetPswd.js:29]: ok, that's fine!

File: frontend/src/App.js [hunk 5]
```diff
@@ -1,24 +1,19 @@
...
-        </a>
-      </header>
-    </div>
+    <>
+    <Routes>
+      <Route path = "/" element = {<FormPage/>}></Route>
+      <Route path = "/reset-username-form" element = {<ResetUsernameForm/>} ></Route>
+      <Route path = "/reset-password-form" element = {<ResetPasswordForm/>} ></Route>
+    </Routes>
+    </>
```
Comment (kdata12) [inline frontend/src/App.js:20]: nice!

File: frontend/src/form.module.css [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    height: 99vh;
+    width: 50vh;
+    border-radius: 15px;
+    box-shadow: 4px 4px 10px gray;
+
+}
+
+.sidebar_btn {
+
+}
```
Comment (kdata12) [inline frontend/src/form.module.css:79]: remove if not used

**Review by kdata12 — APPROVED** [review]
lgtm!

**Pull request merged**